docs/IncrementalBuild.md

```diff
@@ -0,0 +1,30 @@
+Incremental Build
+=================
+
+This repository can build the entire ASP.NET Core stack from scratch. However, in most cases, we only want to build part of the stack.
+
+## Building incrementally
```

Comment:
I'm not very clear on what the workflow for this is (plus a little unsure as to why we're conflating the two topics here). Is the intent than an ordinary build of Universe completely no-ops if it has boms files checked in?

Reply:
"Incremental" is a poor word choice. I'll change this to `Building with cached artifacts`.

The workflow you described is exactly what we're after. Only build when we need to. If everything in universe is built and available from a cache, why rebuild?